python-packages/smithy-http/smithy_http/aio/crt.py

```diff
@@ -301,23 +300,14 @@ async def _marshal_request(
             # If the body is already directly in memory, wrap in a BytesIO to hand
             # off to CRT.
             crt_body = BytesIO(body)
+        elif not self._is_sync_stream(body):
+            # If the body is an async stream.... read it all into memory. This is
+            # very unfortunate, but necessary because the CRT doesn't currently
+            # have the capability to read async. We will likely have to implment
+            # this capability into it ourselves, or implment a thread-based wrapper.
+            crt_body = BytesIO(await request.consume_body_async())
```

Comment:
I wonder if we can do anything like `readinto` with a buffer for the CRT to avoid having to do this all at once. I'll reach out to them to see if there's anything quick we can do here.

Reply:
I mean we could ostensibly pass them an `InputStream` that does its own thing, but at the end of the day it's synchronous.